coderd/userauth.go

```diff
@@ -320,6 +320,22 @@ func (api *API) loginRequest(ctx context.Context, rw http.ResponseWriter, req co
 		return user, database.GetAuthorizationUserRolesRow{}, false
 	}
 
+	if user.Status == database.UserStatusDormant {
+		//nolint:gocritic // System needs to update status of the user account (dormant -> active).
+		user, err = api.Database.UpdateUserStatus(dbauthz.AsSystemRestricted(ctx), database.UpdateUserStatusParams{
+			ID:        user.ID,
+			Status:    database.UserStatusActive,
+			UpdatedAt: database.Now(),
+		})
+		if err != nil {
+			logger.Error(ctx, "unable to update user status to active", slog.Error(err))
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "Internal error.",
```

Comment:
"Internal error" doesn't give the user any useful indication of the next step. Suggest something along the lines of "Try again later, or contact an admin for assistance."

Reply:
What happens when a dormant user attempts to login when the active seat limit has been reached?